test/extended/openstack/servergroup.go

```diff
@@ -6,80 +6,114 @@ import (
+				ms := dc.Resource(schema.GroupVersionResource{
+					Group:    "machine.openshift.io",
+					Resource: "machines",
+					Version:  "v1beta1",
+				})
+				for _, item := range nodeList.Items {
+					uuid := strings.TrimPrefix(item.Spec.ProviderID, "openstack:///")
+					masterInstanceUUIDs = append(masterInstanceUUIDs, uuid)
+
+					machineAnnotation := strings.SplitN(item.Annotations["machine.openshift.io/machine"], "/", 2)
+					o.Expect(machineAnnotation).To(o.HaveLen(2))
+
+					res, err := ms.Namespace(machineAnnotation[0]).Get(ctx, machineAnnotation[1], metav1.GetOptions{})
+					o.Expect(err).NotTo(o.HaveOccurred())
+
+					instanceServerGroupNameField := getFromUnstructured(res, "spec", "providerSpec", "value", "serverGroupName")
+					o.Expect(instanceServerGroupNameField).NotTo(o.BeNil(), "the Server group name should be present in the Machine definition")
+
+					instanceServerGroupName := instanceServerGroupNameField.(string)
+					o.Expect(instanceServerGroupName).NotTo(o.BeEmpty(), "the Server group name should not be the empty string")
+					if serverGroupName == "" {
+						serverGroupName = instanceServerGroupName
+					} else {
+						o.Expect(instanceServerGroupName).To(o.Equal(serverGroupName), "two Control plane Machines have different serverGroupName set")
+					}
+				}
+			}
+		})
 
-				res, err := ms.Namespace(machineAnnotation[0]).Get(ctx, machineAnnotation[1], metav1.GetOptions{})
+		g.It("creates Control plane nodes in a server group", func() {
+			g.By("checking the actual members of the Server group")
+			{
+				computeClient, err := client(serviceCompute)
 				o.Expect(err).NotTo(o.HaveOccurred())
 
-				instanceServerGroupNameField := getFromUnstructured(res, "spec", "providerSpec", "value", "serverGroupName")
-				o.Expect(instanceServerGroupNameField).NotTo(o.BeNil(), "the Server group name should be present in the Machine definition")
+				serverGroupsWithThatName, err := serverGroupIDsFromName(computeClient, serverGroupName)
+				o.Expect(serverGroupsWithThatName, err).To(o.HaveLen(1), "the server group name either was not found or is not unique")
 
-				instanceServerGroupName := instanceServerGroupNameField.(string)
-				o.Expect(instanceServerGroupName).NotTo(o.BeEmpty(), "the Server group name should not be the empty string")
-				if serverGroupName == "" {
-					serverGroupName = instanceServerGroupName
-				} else {
-					o.Expect(instanceServerGroupName).To(o.Equal(serverGroupName), "two Control plane Machines have different serverGroupName set")
-				}
+				serverGroup, err := servergroups.Get(computeClient, serverGroupsWithThatName[0]).Extract()
+				o.Expect(serverGroup.Members, err).To(o.ContainElements(masterInstanceUUIDs...))
 			}
-		}
+		})
 
-		g.By("checking the actual members of the Server group")
-		{
-			computeClient, err := client(serviceCompute)
+		g.It("creates Control plane nodes on separate hosts when serverGroupPolicy is anti-affinity", func() {
```

Comment:
Please add a comment stating the minimum version and the Jira or BZ that implemented/fixed it. In this case, that should be respectively 4.10 and https://issues.redhat.com/browse/OSASINFRA-2507

Reply:
Done